validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java

```diff
@@ -132,6 +134,35 @@ public synchronized PostKeyResult loadMutableValidator(
     return PostKeyResult.success();
   }
 
+  public synchronized PostKeyResult loadExternalMutableValidator(
+      final BLSPublicKey publicKey,
+      final URL signerUrl,
+      final Optional<SlashingProtectionImporter> slashingProtectionImporter) {
+    if (!canAddValidator()) {
+      return PostKeyResult.error("Not able to add validator");
+    }
+
+    if (slashingProtectionImporter.isPresent()) {
+      final Optional<String> errorString =
+              slashingProtectionImporter.get().updateSigningRecord(publicKey, LOG::debug);
+      if (errorString.isPresent()) {
+        return PostKeyResult.error(errorString.get());
+      }
+    }
+    if (ownedValidators.hasValidator(publicKey)) {
+      return PostKeyResult.duplicate();
+    }
+
+    final AddValidatorResult validatorAddResult =
+            mutableValidatorSource.get().addValidator(publicKey, signerUrl);
```

Comment:
This will need the separate `mutableExternalValidatorSource` which would also be declared as type `Optional<ValidatorSource>` but the implementation would be a `ExternalValidatorSource`.  Otherwise we'll try to add an external validator to `LocalValidatorSource` which will throw `UnsupportedOperationException`.